include/clap/helpers/host.hh

```diff
@@ -0,0 +1,176 @@
+      // Methods to override //
+      /////////////////////////
+
+      // clap_host
+      virtual void requestRestart() noexcept {}
+      virtual void requestProcess() noexcept {}
+      virtual void requestCallback() noexcept {}
+
+      // clap_host_audio_ports
+      virtual bool implementsAudioPorts() const noexcept { return false; }
+      bool audioPortsIsRescanFlagSupported(uint32_t flag) noexcept { return false; }
+      void audioPortsRescan(uint32_t flags) noexcept {}
+
+      // clap_host_gui
+      virtual bool implementsGui() const noexcept { return false; }
+      virtual void guiResizeHintsChanged() noexcept {}
+      virtual bool guiRequestResize(uint32_t width, uint32_t height) noexcept { return false; }
+      virtual bool guiRequestShow() noexcept { return false; }
+      virtual bool guiRequestHide() noexcept { return false; }
+      virtual void guiClosed(bool wasDestroyed) noexcept {}
+
+      // clap_host_latency
+      virtual bool implementsLatency() const noexcept { return false; }
+      static void latencyChanged() noexcept {}
+
+      // clap_host_log
+      virtual bool implementsLog() const noexcept { return false; }
+      virtual void logLog(clap_log_severity severity, const char *message) const noexcept {}
+
+      // clap_host_params
+      virtual bool implementsParams() const noexcept { return false; }
+      virtual void paramsRescan(clap_param_rescan_flags flags) noexcept {}
+      virtual void paramsClear(clap_id paramId, clap_param_clear_flags flags) noexcept {}
+      virtual void paramsRequestFlush() noexcept {}
+
+      // clap_host_posix_fd_support
+      virtual bool implementsPosixFdSupport() const noexcept { return false; }
+      virtual bool posixFdSupportRegisterFd(int fd, clap_posix_fd_flags_t flags) noexcept { return false; }
+      virtual bool posixFdSupportModifyFd(int fd, clap_posix_fd_flags_t flags) noexcept { return false; }
+      virtual bool posixFdSupportUnregisterFd(int fd) noexcept { return false; }
+
+      // clap_host_remote_controls
+      virtual bool implementsRemoteControls() const noexcept { return false; }
+      virtual void remoteControlsChanged() noexcept {}
+      virtual void remoteControlsSuggestPage(clap_id pageId) noexcept {}
+
+      // clap_host_state
+      virtual bool implementsState() const noexcept { return false; }
+      virtual void stateMarkDirty() noexcept {}
+
+      // clap_host_timer_support
+      virtual bool implementsTimerSupport() const noexcept { return false; }
+      virtual bool timerSupportRegisterTimer(uint32_t periodMs, clap_id *timerId) noexcept { return false; }
+      virtual bool timerSupportUnregisterTimer(clap_id timerId) noexcept { return false; }
+
+      // clap_host_tail
+      virtual bool implementsTail() const noexcept { return false; }
+      static void tailChanged() noexcept {}
+
+      // clap_host_thread_check
+      virtual bool implementsThreadCheck() const noexcept { return false; }
```

Comment:
I'd make the thread check implementation mandatory maybe? :thinking:

Reply:
It isn't wrong to have it optional, but you loose 95% of the checks if you don't have the thread check. I think it isn't reasonable to start implementing a host without it.
Please make `threadCheckIsMainThread()` and `threadCheckIsAudioThread()` pure virtual and get rid of `implementsThreadCheck()`.